gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/MostlyMySqlDagManagementStateStore.java

```diff
@@ -54,18 +56,30 @@ public class MostlyMySqlDagManagementStateStore implements DagManagementStateSto
   // dagToJobs holds a map of dagId to running jobs of that dag
   private final Map<DagManager.DagId, LinkedList<Dag.DagNode<JobExecutionPlan>>> dagToJobs = new ConcurrentHashMap<>();
   private final Map<DagManager.DagId, Long> dagToDeadline = new ConcurrentHashMap<>();
-  private final DagStateStore dagStateStore;
-  private final DagStateStore failedDagStateStore;
+  private DagStateStore dagStateStore;
+  private DagStateStore failedDagStateStore;
+  private boolean dagStoresInitialized = false;
```

Comment:
`dagStoresInitialized` signify that the dag stores have been initialized. It has nothing to do with being a leader.
Note that dag stores cannot be initialized in constructor because topology map is not available at that time.

Reply:
can you add the last part in a comment? it's helpful to document the decision behind not initializing the stores in constructor and needing this var.